Fixed crash if color picker left open when control is deallocated.

Dejal opened the pull request:
For example when switching to another view.  The color picker is now
automatically closed, which is a better experience, and avoids the
crash.

Dejal committed:
Fixed crash if color picker left open when control is deallocated.

For example when switching to another view.  The color picker is now
automatically closed, which is a better experience, and avoids the
crash.

DarkDust commented:
Removing the notification observer is correct, but we should only reset the panel's target if `panel.target == self`. Someone might have changed the target of the shared panel.

Dejal commented:
There's no getter for the `target` property, so there's no way to check its value.  Elsewhere in the code sets it to `nil`.  Though it's weak linked, so probably don't actually need to reset it.

Dejal commented:
Actually, not sure about the weak linking.

DarkDust commented:
Oops, it really only is a setter without a getter. How strange. I seems to be weak, just ran a quick test and it didn't increase the retain count of my object (it if weren't, `dealloc` of DFColorWell wouldn't be called, would it?)

Dejal committed:
Tweaked fix of crash if color picker left open when control is deallocated to not clear target & action.

Dejal commented:
Strange indeed.  I agree that it seems to be weak, so I've removed those two lines.  Thanks for the feedback.

Dejal committed:
Added gesture recognizer support

The mouse event doesn't work inside a table view.  Not sure why, but
using a gesture recognizer works around the issue.

Dejal committed:
Merge remote-tracking branch 'danieljfarrell/master'

danieljfarrell commented:
Hmmm mouse events don't work inside the table view. Would making the table view the first responder solve this problem?

Dejal commented:
Other mouse events work if you set the highlight mode to None.  But the -mouseDown: doesn't work, hence my workaround.

Dejal committed:
Avoid a crash when rendering in IB.

DarkDust commented:
Isn't this already solved by merging #10? Or is there something that still needs to be fixed?

Dejal committed:
Merge remote-tracking branch 'danieljfarrell/master'

danieljfarrell commented:
I will take a look. There are merge conflict.

Dejal committed:
Avoid sending the action if we are setting the default color.

Fixes an issue on High Sierra where the color became white every time
the control is loaded, as the default color was changing the app's
settings before it could populate the color well.

Dejal commented:
Sorry, did I accidentally trigger a pull request?  I was fixing an issue on [my fork of the project](https://github.com/danieljfarrell/DFColorWell/compare/master...Dejal:master): sending the action when setting the default color was overriding my app's settings on High Sierra. See my branch for my changes.
